Automattic/jetpack
# Docker: Migrate from MySQL to MariaDB to allow for better Apple M1 compatibility

This PR is a redo of #19237

This change migrates the database engine from MySQL to MariaDB, because MySQL images do not support the arm/v8 architecture the new Apple Sillicon computers use.

Seems that the MySQL images built on Docker Hub are only for the `amd64` architecture and can't run on the ARM architecture. 

MariaDB on the other hand has builds for more architectures and it should be a drop-in replacement for MySQL.

#### Changes proposed in this Pull Request:

* Upgrade the Docker database image from MySQL to MariaDB

#### Jetpack product discussion

N/a

#### Does this pull request change what data or activity we track or use?

N/a

#### Testing instructions:

* Stop Docker stack gracefully - either CTRL+C the running `jetpack docker up` (if it's not started in the detached state with `-d` argument) or run `jetpack docker down`
  * This is required, because the migration to MariaDB can't happen if the database is crashed 
* Backup your database folder from `tools/docker/data/jetpack_dev_mysql` 
* Apply branch
* `jetpack docker up`
* Monitor the logs for errors and see if the DB image didn't manage to start (it should start)
* Monitor the logs and see if the migration was automatically performed correctly and no errors were shown.
* Make sure your test site is working
